crates/iota-indexer/src/test_utils.rs

```diff
@@ -42,64 +42,51 @@ impl IndexerTypeConfig {
 }
 
 pub async fn start_test_indexer(
-    db_url: Option<String>,
+    db_url: String,
+    drop_and_create_db: bool,
+    reset_db: bool,
     rpc_url: String,
     reader_writer_config: IndexerTypeConfig,
     data_ingestion_path: Option<PathBuf>,
-    new_database: Option<&str>,
 ) -> (PgIndexerStore, JoinHandle<Result<(), IndexerError>>) {
     start_test_indexer_impl(
         db_url,
+        drop_and_create_db,
+        reset_db,
         rpc_url,
         reader_writer_config,
-        // reset_database
-        false,
         data_ingestion_path,
         CancellationToken::new(),
-        new_database,
     )
     .await
 }
 
 /// Starts an indexer reader or writer for testing depending on the
-/// `reader_writer_config`. If `reset_database` is true, the database instance
-/// named in `db_url` will be dropped and reinstantiated.
+/// `reader_writer_config`.
 pub async fn start_test_indexer_impl(
-    db_url: Option<String>,
+    db_url: String,
+    drop_and_create_db: bool,
+    reset_db: bool,
```

Comment:
These are not independent I reckon. If I pass 
```rust
drop_and_create_db: true,
reset_db: false
```
A  new database will be created according to the `db_url`, but no migrations will be run, and hence we will still have an issue.

So I would propose to keep only a `reset_db` that abstracts away the two internal requirements for a new db. These are: recreate the database, apply migrations.

If `reset_db` is `false` then the database in `db_url` will remain intact.

Reply:
Good point, I see. Let me adjust this!

Reply:
Fixed with https://github.com/iotaledger/iota/pull/5590/commits/c57ea7e3dc0e4fe7bd2f6d2ef647e6bc2e180de0